trtools/dumpSTR/dumpSTR.py

```diff
@@ -514,102 +524,167 @@ def GetAllCallFilters(call_filters):
         reasons.append(filt.name)
     return reasons
 
-def FilterCall(sample, call_filters):
-    r"""Apply call-level filters and return filter reason.
 
-    Parameters
-    ----------
-    sample : vcf._Call
-       The call to be filtered
-    call_filters : list of filters.Reason
-       List of call-level filters
+_NOCALL_INT_FORMAT_VAL = -2147483648
 
-    Returns
-    -------
-    reasons : list of str
-       List of string description of fitler reasons
-    """
-    reasons = []
-    for cfilt in call_filters:
-        if cfilt(sample) is not None: reasons.append(cfilt.GetReason())
-    return reasons
 
-def ApplyCallFilters(record, reader, call_filters, sample_info):
-    r"""Apply call-level filters to a record
+def ApplyCallFilters(record: trh.TRRecord,
+                     call_filters: List[filters.FilterBase],
+                     sample_info: Dict[str, np.ndarray],
+                     sample_names: List[str]) -> trh.TRRecord:
+    r"""Apply call-level filters to a record.
 
-    Returns a new record with FILTER populated for each sample.
+    Returns a TRRecord object with the FILTER (or DUMPSTR_FILTER)
+    format field updated for each sample.
     Also updates sample_info with sample level stats
 
     Parameters
     ----------
-    record : vcf._Record
-       The record to apply filters to
-    reader : vcf.Reader
-       The vcf.Reader object we're reading from
-    call_filters : list of filters.Reason
+    record :
+       The record to apply filters to.
+       Note: once this method has been run, this object
+       will be in an inconsistent state. All further use
+       should be directed towards the returned TRRecord object.
```

Comment:
That'd be a fair bit of hassle - the TRRecord constructor is quite particular so cloning it would have to have a lot of checks to figure out which optional arguments to pass through. The idea is a good one, I'd do it if it weren't a pain, but given that the code works as is and we don't expect end users to call this function I think I'd like to pass on it.

Reply:
Rethinking my response:

The issue is that it's not the trh.TRRecord object that's being modified but rather the cyvcf2.Variant object that it contains, which is a necessary step in using cyvcf2.Writer. The appropriate thing to do would be to clone the underlying cyvcf2.Variant object and then work with the clone, which is a good idea and not something I thought of. Unfortunately, I just checked and that cyvcf2.Variant doesn't have a clone method. So this is a no go - we have to modify the original cyvcf2.Variant, and that necessarily puts the original wrapping TRRecord in an inconsistent state.

The only other fix would be to modify the original wrapping TRRecord to fix it, but that seems like it would be very error-prone code. Creating a new TRRecord that's bound to be in a clean state seems better, and that's what the code currently does.

Reply:
ok that's fair. I thought it would be simple, but it it's very involved, we don't need to do it. It's unlikely that we need the original object down the line anyway.